src/librustc_typeck/check/method/suggest.rs

```diff
@@ -173,7 +173,18 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                                      if mode == Mode::MethodCall {
                                          "method"
                                      } else {
-                                         "associated item"
+                                         match item_name.as_str().chars().next() {
+                                             Some(name) => {
+                                                 if name.is_lowercase() {
+                                                     "function or associated item"
```

Comment:
We don't know if it's a function call at this stage. This fix does not introduce that checking, though @photoszzt may work on that next if he has time.